routing.go

```diff
@@ -0,0 +1,64 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	routing "github.com/libp2p/go-libp2p-routing"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// RoutingDiscovery is an implementation of discovery using ContentRouting
+type RoutingDiscovery struct {
+	router routing.ContentRouting
+}
+
+func NewRoutingDiscovery(router routing.ContentRouting) Discovery {
```

Comment:
I'd just return a `RoutingDiscovery` or a `*RoutingDiscovery` ("accept interfaces, return structs" tends to be considered "best practice").

Reply:
sure, let's follow them!